v1/dictionary/handlers.js

```diff
@@ -0,0 +1,133 @@
+var wd = require('word-definition');
+
+require('kettle');
+
+
+/* General function to get definition 
+ * from all the different third-party dictionary services being used
+ * Simply gives the complete response from the third party service 
+ * Does not manipulate it in any way
+ * The function returns a promise whose result contains the response
+ * from the third-party service
+ * Service codes -
+ *  Wiktionary -> 'wiki'
+ */
+function definition(service, lang, word) {
+
+  //Wiktionary Service
+  if(service == 'wiki') {
+    var promise = fluid.promise();
+    wd.getDef(word, lang, null, function(data) {
+      promise.resolve(data);
+    });
+    return promise;
+  }
+}
+
+/* General function to catch the errors
+ * From the third-party dictionary services
+ * Service codes -
+ *  Wiktionary -> 'wiki'
+ */
+function checkDictionaryErr(service, serviceResponse) {
+  //Wiktionary Service
+  if(service == 'wiki') {
+
+    //Check if there is an error
+    if(serviceResponse.err) {
+
+      //Word not found
+      if(serviceResponse.err == 'not found') {
+        return {
+          statusCode: 404,
+          errorMessage: 'Dictionary Service: Word not found (Wiktionary)'
+        }
+      }
+
+      //Language unsupported by the third-party service
+      else if(serviceResponse.err == 'unsupported language') {
+        return {
+          statusCode: 400,
+          errorMessage: 'Dictionary Service: Unsupported Language (Wiktionary)'
+        }
+      }
+
+      //Default return object when error hasn't been handled yet
+      else return {
+        statusCode: 500,
+        errorMessage: 'Dictionary Service: The error hasn\'t been handled yet (Wiktionary)'
+      };
+    }
+
+    //Return false if no error found
+    else return;
+  }
+}
+
+//Main request handler - Gives only the definition of the word
+fluid.defaults('AdaptiveContentServices.Dictionary.serverConfig.mainDictionaryHandler', {
+  gradeNames: 'kettle.request.http',
+  invokers: {
+    handleRequest: 'AdaptiveContentServices.Dictionary.serverConfig.mainDictionaryRequestHandler'
+  }
+});
+
+AdaptiveContentServices.Dictionary.serverConfig.mainDictionaryRequestHandler = function(request) {
+  try {
+    var version = request.req.params.version;
+    var word = request.req.params.word;
+    var lang = request.req.params.language;
+
+    var serviceResponse, errorContent;
+
+    definition('wiki', lang, word)
+    .then(
+      function(result) {
+        serviceResponse = result;
+
+        errorContent = checkDictionaryErr('wiki', serviceResponse);
+
+        request.res.set({
+          'Content-Type': 'application/json',
+          'Access-Control-Allow-Origin': '*',
+          'Access-Control-Allow-Headers': 'Origin, X-Requested-With, Content-Type, Accept'
+        });
+    
+        //Error Responses
+        if(errorContent) {
+          request.res.status(errorContent.statusCode)
+          .send({
```

Comment:
Please implement sending responses and errors using the standard Kettle request events (`onSuccess` and `onError`) described at https://github.com/fluid-project/kettle/blob/master/docs/RequestHandlersAndApps.md#how-to-implement-a-request-handler, rather than directly using "raw" `.send` - the event-driven approach is necessary for handlers to work in the larger context of Kettle applications and with things like Kettle's middleware implementation.

We haven't talked about Infusion events much but they're another important building block of the framework, described in some detail at https://docs.fluidproject.org/infusion/development/InfusionEventSystem.html, and Kettle makes a lot of use of them.

Reply:
I had first used the `onSuccess` and `onError` request events but that limited my response status code to `200` and `404`, although the JSON response will have the correct status code.
Is there a way to specifically set the status code for these events?

Reply:
For `onError`, you can do the following: 

```
    request.events.error.fire({
        message: "GET request received on path /handlerPath",
        statusCode: 500
    });
```

Reply:
Okay :+1: 
Thanks! 
I will make these changes.